### PR TITLE
DOC:  fft: add note about FHT formulas

### LINKED ISSUE #19573: scipy.fft.fht - documentation issue
### Describe your issue.

In your published documentation for scipy.fft.fht (https://docs.scipy.org/doc/scipy/reference/generated/scipy.fft.fht.html) and scipy.fft.ifht, the formulas you report for the forward and inverse Hankel transforms appear incorrect. For the forward transform, I believe the integral kernel should be written as "a(r) J_u(kr) r dr". For the inverse transform, I believe the integral kernel should be written as "A(k) J_u(kr) k dk". See, for example, the link here:
https://mathworld.wolfram.com/HankelTransform.html
I believe this is just a typo in the documentation but wanted to alert you of it. I also wanted to confirm that the algorithm you developed was based around the correct formula.

### Reproducing Code Example

```python
issue does not concern code
```

### Error message

```shell
issue does not concern code
```

### SciPy/NumPy/Python version and system information

```shell
issue does not concern code
```

### PR DESCRIPTION
**Pull request by willtryagain**
[docs only]

### Reference issue
Closes https://github.com/scipy/scipy/issues/19573

### CODE REVIEW

**Comment by willtryagain** [comment]
apologies, I forgot to add [docs only] in the commit message.

**Comment by j-bowhay** [comment]
@ntessore would you be interested in reviewing?

**Comment by ntessore** [comment]
Sure, no problem.

@willtryagain, I don't think the text I wrote in the other issue is a good fit for the docstring. I think a short acknowledgement of the different definition might work better:

    where :math:`J_\mu` is the Bessel function of order :math:`\mu`.  The index
    :math:`\mu` may be any real number, positive or negative.  Note that the
    numerical Hankel transform uses an integrand of :math:`k dr`, while the
    mathematical Hankel transform is commonly defined using :math:`r dr`.

**Review by ntessore — CHANGES_REQUESTED** [review]
Thanks @willtryagain, I think this reads better for a compact reference. Could you also add a similar sentence to `ifht`?

File: scipy/fft/_fftlog.py [hunk 1]
```diff
@@ -54,7 +54,9 @@ def fht(a, dln, mu, offset=0.0, bias=0.0):
         A(k) = \int_{0}^{\infty} \! a(r) \, J_\mu(kr) \, k \, dr \;,
 
     where :math:`J_\mu` is the Bessel function of order :math:`\mu`.  The index
-    :math:`\mu` may be any real number, positive or negative.
+    :math:`\mu` may be any real number, positive or negative.  Note that the
+    numerical Hankel transform uses an integrand of :math:`k dr`, while the
+    mathematical Hankel transform is commonly defined using :math:`r dr`.
```
Comment (ntessore) [inline scipy/fft/_fftlog.py:59]:
```suggestion
    numerical Hankel transform uses an integrand of :math:`k \, dr`, while the
    mathematical Hankel transform is commonly defined using :math:`r \, dr`.
```

File: scipy/fft/_fftlog.py [hunk 2]
```diff
@@ -212,7 +214,9 @@ def ifht(A, dln, mu, offset=0.0, bias=0.0):
         a(r) = \int_{0}^{\infty} \! A(k) \, J_\mu(kr) \, r \, dk \;,
 
     where :math:`J_\mu` is the Bessel function of order :math:`\mu`.  The index
-    :math:`\mu` may be any real number, positive or negative.
+    :math:`\mu` may be any real number, positive or negative. Note that the
+    numerical inverse Hankel transform uses an integrand of :math:`r dk`, while the
+    mathematical inverse Hankel transform is commonly defined using :math:`k dk`.
```
Comment (ntessore) [inline scipy/fft/_fftlog.py:219]:
```suggestion
    numerical inverse Hankel transform uses an integrand of :math:`r \, dk`, while the
    mathematical inverse Hankel transform is commonly defined using :math:`k \, dk`.
```

**Review by ntessore — CHANGES_REQUESTED** [review]
Sorry about this further request! I could not find a docs preview anywhere on here. After checking locally, I realised that the spacing in the math didn't look right.

**Comment by lucascolley** [comment]
For future reference, the docs preview is available under the CI job "Check the rendered docs here!"

**Review by ntessore — APPROVED** [review]

**Review by lucascolley — APPROVED** [review]
thanks for the review @ntessore ! 

@willtryagain would you mind editing the title of the PR to something slightly more specific, perhaps "DOC: fft: add note about FHT formulas"? 'Documentation issue' is a little vague. This will make it slightly quicker for whoever squash-merges the PR.

Also, in the future, if you could write commit messages roughly following [the guidance in our docs](http://scipy.github.io/devdocs/dev/contributor/development_workflow.html#writing-the-commit-message), that would be great! This isn't necessary at all for complicated PRs with lots of commits, as they will be squash-merged and the title can be edited, but for simple PRs which might be ready to merge straight away this can help us :)

EDIT: if anyone notices this and wants to, feel free to fix my typo from 'formulas' to 'formulae' :)

**Comment by willtryagain** [comment]
thank you. your patience and guidance is much appreciated. I would follow the guidelines in my next PRs.

**Review by j-bowhay — APPROVED** [review]
Thanks everyone

**Pull request merged**